src/zkp/ecdsa_adaptor.rs

```diff
@@ -84,31 +88,116 @@ impl EcdsaAdaptorSignature {
     }
 }
 
+fn encrypt_helper<C: Signing>(
```

Comment:
If we have such kind of helper functions, I would prefer them to be defined further down from where they are used. Makes it easier to scan through the code if the public API is defined first and followed by code that is only an implementation detail.

In this case though, I don't think we need a helper function at all. Instead, I think it is clearer if we prefer forwarding to other public functions (see other comment) plus for the no_aux usecase, just duplicate the invocation. DRY has a price in the form of indirection and in this case, I don't think it is worth it :)

Reply:
Agreed, updated as suggested. Another solution I thought of would be to have a single function `encrypt_no_std` that takes an `Option<&[u8; 32]>` as pasted below but I don't know if the match would have some negative performance impact.

```rust
    pub fn encrypt_no_std<C: Signing>(
        secp: &Secp256k1<C>,
        msg: &Message,
        sk: &SecretKey,
        enckey: &PublicKey,
        aux_rand: Option<&[u8; 32]>,
    ) -> EcdsaAdaptorSignature {
        let mut adaptor_sig = ffi::EcdsaAdaptorSignature::new();

        let aux_rand_ptr = match aux_rand {
            Some(aux) => aux.as_c_ptr() as *mut ffi::types::c_void,
            None => ptr::null_mut(),
        };

        unsafe {
            debug_assert!(
                ffi::secp256k1_ecdsa_adaptor_encrypt(
                    *secp.ctx(),
                    &mut adaptor_sig,
                    sk.as_c_ptr(),
                    enckey.as_c_ptr(),
                    msg.as_c_ptr(),
                    ffi::secp256k1_nonce_function_ecdsa_adaptor,
                    aux_rand_ptr,
                ) == 1
            );
        };
        EcdsaAdaptorSignature(adaptor_sig)
    }
```